app/src/main/java/com/firebase/uidemo/database/ChatIndexActivity.java

```diff
@@ -34,12 +35,13 @@ public void onClick(View v) {
 
     @Override
     protected FirebaseRecyclerAdapter<Chat, ChatHolder> getAdapter() {
-        return new FirebaseIndexRecyclerAdapter<Chat, ChatHolder>(
-                Chat.class,
+        return new FirebaseRecyclerAdapter<Chat, ChatHolder>(
+                new FirebaseIndexArray<>(
```

Comment:
I think if we merge this PR we will recommend that everyone use the `LifecycleOwner` version of the adapters, but I wouldn't want those people to also have to understand `FirebaseArray`.  So we either need to do all the overloads or find some other way to get the owner in there.